core/server/services/users.js

```diff
@@ -17,17 +17,59 @@ const path = require('path');
  * @prop {Object} ApiKey.NotFoundError
  * @prop {Object} User
  * @prop {(Object) => Promise} User.destroy
+ * @prop {(Object) => Promise} User.findAll
+ * @prop {Object} Session
+ * @prop {(Object) => Promise} Session.findAll
+ */
+
+/**
+ * @typedef {Object} IAuth
+ * @prop {Object} setup
+ * @prop {(isComplete: boolean) => () => Promise} setup.assertSetupCompleted
+ * @prop {Object} passwordreset
+ * @prop {(email: string, apiSettings: Object, transcation?: Object) => Promise<string>} passwordreset.generateToken
+ * @prop {(token: string, apiMail: Object) => Promise} passwordreset.sendResetNotification
  */
 
 class Users {
     /**
      * @param {Object} dependencies
      * @param {IdbBackup} dependencies.dbBackup
      * @param {IModels} dependencies.models
+     * @param {IAuth} dependencies.auth
+     * @param {Object} dependencies.apiMail
+     * @param {Object} dependencies.apiSettings
      */
-    constructor({dbBackup, models}) {
+    constructor({dbBackup, models, auth, apiMail, apiSettings}) {
         this.dbBackup = dbBackup;
         this.models = models;
+        this.auth = auth;
+        this.apiMail = apiMail;
+        this.apiSettings = apiSettings;
+    }
+
+    async resetAllPasswords(frameOptions) {
+        return this.models.Base.transaction(async (t) => {
+            frameOptions.transacting = t;
+
+            // Reset all passwords
+            const users = await this.models.User.findAll({...frameOptions});
```

Comment:
What's the difference of using `...` spread operator to just doing a `.findAll(frameOptions)` call?

Reply:
I've fixed it, I guess I was expecting to add more properties to the findAll configuration object but didn't end up doing so.